pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/typedefinition/JavaTypeDefinition.java

```diff
@@ -37,9 +40,13 @@ public static JavaTypeDefinition forClass(TypeDefinitionType type, JavaTypeDefin
             }
         case UPPER_BOUND:
         case UPPER_WILDCARD:
-            return new JavaTypeDefinitionUpper(type, intersectionTypes);
+            // In theory, if one of the bounds can't be resolved, then the type is useless.
+            // Looking at the implementation of JavaTypeDefinitionUpper, it looks like only the
+            // first bound is used, so we could only check for the first array component.
+            // But isn't that behaviour weird ? Where are the other bounds useful then ?
+            return ArrayUtils.contains(intersectionTypes, null) ? null : new JavaTypeDefinitionUpper(type, intersectionTypes);
         case LOWER_WILDCARD:
-            return new JavaTypeDefinitionLower(intersectionTypes);
+            return ArrayUtils.contains(intersectionTypes, null) ? null : new JavaTypeDefinitionLower(intersectionTypes);
```

Comment:
I'm not sure I understand these changes...

Reply:
If one of the intersection bounds wasn't resolved (is null), we previously returned a type definition containing null. That type definition would have been treated as resolved (since it's not null), though any call to it (such as isArrayType) would have failed with an NPE later on. This change fails safe by returning null early on, to avoid late NPEs